src/Board.py

```diff
@@ -355,66 +359,76 @@ def getAlgebraicNotationOfMoveWithFileAndRank(self, move, short=True):
         notation += self.positionToHumanCoord(move.newPos)
         return notation
 
+    # TODO this method is never used, remove?
+    @no_type_check
     def humanCoordToPosition(self, coord):
         transTable = str.maketrans('abcdefgh', '12345678')
         coord = coord.translate(transTable)
         coord = [int(c)-1 for c in coord]
         pos = C(coord[0], coord[1])
         return pos
 
-    def positionToHumanCoord(self, pos):
+    def positionToHumanCoord(self, pos: C) -> str:
         transTable = str.maketrans('01234567', 'abcdefgh')
         notation = str(pos[0]).translate(transTable) + str(pos[1]+1)
         return notation
 
-    def isValidPos(self, pos):
+    def isValidPos(self, pos: C) -> bool:
         return 0 <= pos[0] <= 7 and 0 <= pos[1] <= 7
 
-    def getSideOfMove(self, move):
+    def getSideOfMove(self, move: Move) -> bool:
         return move.piece.side
 
+    # TODO this method is never used, remove?
+    @no_type_check
     def getPositionOfPiece(self, piece):
         for y in range(8):
             for x in range(8):
                 if self.boardArray[y][x] is piece:
                     return C(x, 7-y)
 
-    def pieceAtPosition(self, pos):
+    def pieceAtPosition(self, pos: C) -> Piece:  # type: ignore[return] # TODO: add consistent return for else condition
```

Comment:
Do i overlook something or why can't it be `None`?

Reply:
If this is `None` then for example `getAlgebraicNotationOfMoveWithFile()` in `Board.py` will assign `pieceToMove` to `None` which then will be called few lines later with `notation += pieceToMove.stringRep` which will fail as `NoneType` does not have `.stringRep`. Basically Mypy complains a lot about `NoneType` not having `stringRep` without these ignore comments